[WIP] sql: refresh descriptor leases with sqlliveness

jasonmchan opened the pull request:
Previously, descriptor leases were refreshed per-descriptor. This could
be improved by using SQL liveness to heartbeat leases.

This commit is WIP. It introduces liveness-based lease heartbeating. The
system lease table now has a liveness session column. This cannot be
merged because it does not handle mixed-version state.

As a sketch of the implementation, we would need to have an
intermediate version where we fill out both the `sessionID`
and `expiration` columns in the system lease table, to ensure
that nodes running old code respect new leases. In this
intermediate version, when counting leases for a descriptor,
we would check for the validity of both expirations.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/85837)
<!-- Reviewable:end -->

Commit:
sql: refresh descriptor leases with sqlliveness

Previously, descriptor leases were refreshed per-descriptor. This could
be improved by using SQL liveness to heartbeat leases.

This commit is WIP. It introduces liveness-based lease heartbeating. The
system lease table now has a liveness session column. This cannot be
merged because it does not handle mixed-version state.

cockroach-teamcity commented:
[![CLA assistant check](https://cla.crdb.dev/pull/badge/not_signed)](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=85837) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.crdb.dev/cockroachdb/cockroach?pullRequest=85837) before we can accept your contribution.<br/><hr/>**Jason Chan** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.crdb.dev/check/cockroachdb/cockroach?pullRequest=85837) it.</sub>
